core/controllers/retrieve.py

```diff
@@ -0,0 +1,42 @@
+from flask import abort, make_response, request
+from sqlalchemy import select
+
+from core.const import EXCEL_MIMETYPE
+from core.db import db
+from core.db.entities import Programme, ProgrammeProgress, Submission
+
+
+def retrieve():
+    """Handle the download request and return the originally submitted spreadsheet.
+
+    Select file by:
+    - reporting round
+    - programme name
+
+    :return: Flask response object containing the requested spreadsheet.
+    """
+    reporting_round = request.args.get("reporting_round")
+    programme_name = request.args.get("programme_name")
+
+    query = (
+        select(Submission.submission_file)
+        .join(ProgrammeProgress, Submission.id == ProgrammeProgress.submission_id)
+        .join(Programme, ProgrammeProgress.programme_id == Programme.id)
+        .where(Programme.programme_name == programme_name)
```

Comment:
I might be wrong here as it has been a while since I worked on the db, but programme_name isn't unique, and as it's just the name of the place, isn't it possible to have two programmes with the same name in the same round (e.g. "Rotherham" for both TD and HS)? The unique key in Programme is programme_id and not name

Reply:
But even if that's not the case I'm not sure "reporting_round" and "programme_name" necessarily form a composite key

Reply:
Yes, I agree. I'm not fully happy with this way of selecting files, and if we can have duplicate keys I should change it. I was trying to think of how we would like to retrieve these files and how to automate retrieving whole rounds of returns at once. Originally I wanted it to be human readable so I went with place_name and reporting round. Since this isn't going to be user facing prehaps I can just go for programme or submission ID.... Thinking out loud here

Reply:
I think any select statement that expects a single value using a where clause has to use it on a unique key (or something assume that the first value being returned will always be the correct one). 

Any serious db debugging requires us to read the UUIDs anyway, there's not much getting around it.

Reply:
Thanks for this comment it was very informative in deciding how to build this endpoint